### PR TITLE
cleanup more code

### PR DESCRIPTION
**Pull request by wdpypere**

### CODE REVIEW

File: lib/vsc/utils/docs.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def mk_rst_table(titles, columns):
         width = max(map(len, columns[i] + [title]))
 
         # make line template
-        tmpl.append('{%s:{c}<%s}' % (i, width))
+        tmpl.append('{{{}:{{c}}<{}}}'.format(i, width))
```
Comment (stdweird) [inline lib/vsc/utils/docs.py:60]: what is this unreadable crap?

File: lib/vsc/utils/py2vs3/py3.py [hunk 2]
```diff
@@ -36,12 +36,7 @@
 from tempfile import TemporaryDirectory  # noqa
 from urllib.parse import urlencode, unquote  # noqa
 from urllib.request import HTTPError, HTTPSHandler, Request, build_opener, urlopen  # noqa
-try:
-    # py 3.10+
-    from collections import Mapping  # noqa
-except ImportError:
-    # < py 3.10
-    from collections.abc import Mapping  # noqa
+from collections.abc import Mapping # noqa
```
Comment (stdweird) [inline lib/vsc/utils/py2vs3/py3.py:39]: so the comment was wrong?

File: test/exceptions.py [hunk 3]
```diff
@@ -84,18 +82,17 @@ def test_loggedexception_specifiedlogger(self):
...
 
         # if logger is specified, it should be used
         logToFile(tmplog, enable=True)
         self.assertErrorRegex(LoggedException, 'BOOM', raise_loggedexception, 'BOOM', logger=logger1)
         logToFile(tmplog, enable=False)
 
         rootlog = getRootLoggerName()
-        log_re = re.compile("^%s.testlogger_one :: BOOM( \(at .*:[0-9]+ in raise_loggedexception\))?$" % rootlog, re.M)
-        with open(tmplog, 'r') as f:
+        log_re = re.compile(f"^{rootlog}.testlogger_one :: BOOM( \\(at .*:[0-9]+ in raise_loggedexception\\))?$", re.M)
```
Comment (stdweird) [inline test/exceptions.py:92]: use `rf"..` ?

File: test/exceptions.py [hunk 4]
```diff
@@ -107,18 +104,17 @@ def test_loggedexception_callerlogger(self):
...
 
         # if no logger is specified, logger available in calling context should be used
         logToFile(tmplog, enable=True)
         self.assertErrorRegex(LoggedException, 'BOOM', raise_loggedexception, 'BOOM')
         logToFile(tmplog, enable=False)
 
         rootlog = getRootLoggerName()
-        log_re = re.compile("^%s(.testlogger_local)? :: BOOM( \(at .*:[0-9]+ in raise_loggedexception\))?$" % rootlog)
-        with open(tmplog, 'r') as f:
+        log_re = re.compile(f"^{rootlog}(.testlogger_local)? :: BOOM( \\(at .*:[0-9]+ in raise_loggedexception\\))?$")
```
Comment (stdweird) [inline test/exceptions.py:114]: idem, `rf`?

File: test/py2vs3.py [hunk 5]
```diff
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
```
Comment (stdweird) [inline test/py2vs3.py:1]: using the origina module should raise an exception at some point, or at least do a logging.error
and i think almost all these test can be simplified if not removed

File: lib/vsc/utils/py2vs3/py3.py (hunk 2 above)
Comment (wdpypere) [reply lib/vsc/utils/py2vs3/py3.py:39]: it seems so. I can't get it to break. Or it was fixed upstream.

```
[wdpypere@lancelot ~]$ python3.6
Python 3.6.15 (default, Aug  2 2023, 00:00:00) 
[GCC 13.2.1 20230728 (Red Hat 13.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from collections.abc import Mapping
>>> 

[wdpypere@lancelot ~]$ python3.9
Python 3.9.18 (main, Aug 28 2023, 00:00:00) 
[GCC 13.2.1 20230728 (Red Hat 13.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from collections.abc import Mapping
>>> 

[wdpypere@lancelot ~]$ python3.12
Python 3.12.0 (main, Oct  2 2023, 00:00:00) [GCC 13.2.1 20230918 (Red Hat 13.2.1-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from collections.abc import Mapping
>>> 
```

Comment (boegel) [reply lib/vsc/utils/py2vs3/py3.py:39]: https://docs.python.org/3/library/collections.abc.html#collections.abc.Mapping says "New in version 3.5"

File: lib/vsc/utils/docs.py (hunk 1 above)
Comment (wdpypere) [reply lib/vsc/utils/docs.py:60]: to escape a `{` you need to double it.... I'll see if I can come up with some more readable.

Comment (wdpypere) [reply lib/vsc/utils/docs.py]: rewritten with ljust builtin instead of format on format on ...

File: test/py2vs3.py (hunk 5 above)
Comment (wdpypere) [reply test/py2vs3.py:1]: trying to use py2 will already give an import error. I'll add a logging statement when you try to load the py3 version. I think I cleaned most of the code. We could also remove it, but things might break.

Comment (stdweird) [reply test/py2vs3.py:1]: up to you

Comment (wdpypere) [reply test/py2vs3.py:1]: warning added

**Comment by wdpypere** [comment]
@stdweird remarks addressed

**Pull request merged**